WikipediaUnitTests/Code/ArticlePerformanceTests.swift

```diff
@@ -0,0 +1,229 @@
+        setupNetworkStubs()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        LSNocilla.sharedInstance().stop()
+    }
+
+    private func setupNetworkStubs() {
+        
+        guard let dogSummaryJSONData = wmf_bundle().wmf_data(fromContentsOfFile: "DogArticleSummary", ofType: "json"),
+            let dogCollageImageData = wmf_bundle().wmf_data(fromContentsOfFile:"CollageOfNineDogs", ofType:"jpg"),
+            let userGroupsData = wmf_bundle().wmf_data(fromContentsOfFile:"UserInfoGroups", ofType:"json"),
+            let mobileHTMLData = wmf_bundle().wmf_data(fromContentsOfFile:"DogMobileHTML", ofType:"html"),
+            let mobileBaseCSSData = wmf_bundle().wmf_data(fromContentsOfFile:"MobileBase", ofType:"css"),
+            let mobileSiteCSSData = wmf_bundle().wmf_data(fromContentsOfFile:"MobileSite", ofType:"css"),
+            let mobilePCSCSSData = wmf_bundle().wmf_data(fromContentsOfFile:"MobilePCS", ofType:"css"),
+            let mobilePCSJSData = wmf_bundle().wmf_data(fromContentsOfFile:"MobilePCS", ofType:"js"),
+            let i18PCSData = wmf_bundle().wmf_data(fromContentsOfFile:"PCSI18N", ofType: "json"),
+            let redPencilImageData = wmf_bundle().wmf_data(fromContentsOfFile:"RedPencilIcon", ofType:"png"),
+            let commonsLogoImageData = wmf_bundle().wmf_data(fromContentsOfFile:"59pxCommonsLogo.svg", ofType:"webp"),
+            let footerDog1ImageData = wmf_bundle().wmf_data(fromContentsOfFile:"64pxAussieBlacktri", ofType:"jpg"),
+            let footerDog2ImageData = wmf_bundle().wmf_data(fromContentsOfFile:"64pxOkapi2", ofType:"jpg"),
+            let wiktionaryLogoImageData = wmf_bundle().wmf_data(fromContentsOfFile:"54pxWiktionaryLogoV2.svg", ofType:"webp"),
+            let smallCommonsLogoImageData = wmf_bundle().wmf_data(fromContentsOfFile:"40pxCommonsLogo.svg", ofType:"webp"),
+            let wikiQuoteLogoImageData = wmf_bundle().wmf_data(fromContentsOfFile:"46pxWikiquoteLogo.svg", ofType:"webp"),
+            let wikiSpeciesLogoImageData = wmf_bundle().wmf_data(fromContentsOfFile:"46pxWikispeciesLogo.svg", ofType:"webp"),
+            let wikiSourceLogoImageData = wmf_bundle().wmf_data(fromContentsOfFile:"51pxWikisourceLogo.svg", ofType:"webp"),
+            let wikiBooksLogoImageData = wmf_bundle().wmf_data(fromContentsOfFile:"54pxWikibooksLogo.svg", ofType:"webp"),
+            let wikiNewsLogoImageData = wmf_bundle().wmf_data(fromContentsOfFile:"54pxWikinewsLogo.svg", ofType:"webp"),
+            let genericDogImageData = wmf_bundle().wmf_data(fromContentsOfFile:"640pxDogMorphologicalVariation", ofType:"png"),
+            let imageRegex = try? NSRegularExpression(pattern: "https://upload.wikimedia.org/wikipedia/commons/thumb.*", options: []) else {
+            assertionFailure("Error setting up fixtures.")
+            return
+        }
+
+        let _ = stubRequest("GET", "https://en.wikipedia.org/api/rest_v1/page/summary/Dog" as NSString)
```

Comment:
I don't have experience with Nocilla, but just a quick note. If you're finding the `let _ =` unused assignment pattern noisy in these Swift test classes and don't want to do anything gross with Nocilla's `stubRequest` to add Objective-C's unused result attribute to suppress warnings, could be worth adding a simple facade-ish method for the Swift test classes to use. Cleans up the call sites and retains the desired behavior.

Maybe something like:
```swift
@discardableResult
func discardableStubRequest(_ method: String, _ url: LSMatcheable) -> LSStubRequestDSL {
    return stubRequest(method, url)
}
```

Reply:
Great point! I gave this a super quick attempt but it wasn't as simple due to the extra chaining behind most of the stub methods. I'll give it another try tomorrow, but I added the changes welcome label just in case you wanted to take a look too. Nocilla gave me some trouble this round and I notice it hasn't been updated in a long time, so it may be worth looking into other options.